#!/usr/bin/python3
PACKAGE="core_pid_controller"


from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("P", double_t, 0, "Proportional gain when error is positive.", 0,  -100, 100)
gen.add("I", double_t, 0, "Integral gain when error is positive.", 0,  -100, 100)
gen.add("D", double_t, 0, "Derivative gain when error is positive.", 0,  -100, 100)
gen.add("FF", double_t, 0, "Feed forward gain when error is positive.", 0,  -100, 100)

gen.add("use_negative_gains", bool_t, 0, "Whether or not to use negative PIDFF gains", False)

gen.add("neg_P", double_t, 0, "Proportional gain when error is negative.", 0,  -100, 100)
gen.add("neg_I", double_t, 0, "Integral gain when error is negative.", 0,  -100, 100)
gen.add("neg_D", double_t, 0, "Derivative gain when error is negative.", 0,  -100, 100)
gen.add("neg_FF", double_t, 0, "Feed forward gain when error is negative.", 0,  -100, 100)

gen.add("constant", double_t, 0, "Constant control output.", 0, -100, 100)


gen.add("target", double_t, 0, "Target (for debugging only, remove later).", 0,  -20, 20)


exit(gen.generate(PACKAGE, "core_pid_controller", "PID"))
